pkg/ipam-controller/allocator/allocator.go

```diff
@@ -115,76 +93,90 @@ func (pa *poolAllocator) Allocate(ctx context.Context, node string) (nodeAllocat
 		ip.IsBroadcast(endIP, pa.cfg.Subnet) {
 		// out of range
 		log.Info("can't allocate: pool has no free ranges")
-		return nodeAllocationInfo{}, ErrNoFreeRanges
+		return &AllocatedRange{}, ErrNoFreeRanges
 	}
 
 	log.Info("range allocated",
 		"start", startIP, "end", endIP)
-	pa.allocations[node] = allocatedRange{
+	a := AllocatedRange{
 		StartIP: startIP,
 		EndIP:   endIP,
 	}
-	return pa.getNodeAllocationInfo(node, pa.allocations[node]), nil
+	pa.allocations[node] = a
+	return &a, nil
+}
+
+// AllocateFromPool allocates ranges for node
+func (pa *PoolAllocator) AllocateFromPool(ctx context.Context, nodeName string) (*AllocatedRange, error) {
+	allocation, err := pa.allocate(ctx, nodeName)
+	if err != nil {
+		pa.Deallocate(ctx, nodeName)
```

Comment:
Do we need this logic? It looks like that `pa.allocate` returns and error it keeps allocations untouched

Reply:
Removed